src/routes/index.ts

```diff
@@ -0,0 +1,56 @@
+import { LATEST_RELEASES } from '../constants/r2Prefixes';
+import { cached } from '../middleware/cacheMiddleware';
+import { NotFoundMiddleware } from '../middleware/notFoundMiddleware';
+import { OptionsMiddleware } from '../middleware/optionsMiddleware';
+import { OriginMiddleware } from '../middleware/originMiddleware';
+import { R2Middleware } from '../middleware/r2Middleware';
+import { SubtitutionMiddleware } from '../middleware/subtituteMiddleware';
+import { Router } from './router';
+
+export function registerRoutes(router: Router): void {
+  const r2Middleware = new R2Middleware();
+  const originMiddleware = new OriginMiddleware();
+
+  router.options('*', [new OptionsMiddleware()]);
+
+  router.head('/metrics/?:filePath+', [r2Middleware, originMiddleware]);
+  router.get('/metrics/?:filePath+', [cached(r2Middleware), originMiddleware]);
+
+  // Register routes for latest releases (e.g. `/dist/latest/`)
+  for (const branch in LATEST_RELEASES) {
```

Comment:
Interesting, but I believe these should be subrouters, ie how fastify creates groups of routers

And the group is the /:branch/

Reply:
I also wonder if this should be `/dist/:branch` and let the route handler itself handle the correct version stuff.

So instead of foreaching here and creating multiple router handlers, we have only one for each one of these paths (dist, download/release, docs)

Reply:
> Interesting, but I believe these should be subrouters, ie how fastify creates groups of routers

How about we explore subrouters in a later pr just to limit the size and scope of this one?

> So instead of foreaching here and creating multiple router handlers, we have only one for each one of these paths 

Something like this?
```typescript
router.head(
  [`/dist/${branch}`, `/download/release/${branch}`, /* ... */],
  [substitutionMiddleware]
)
```

Reply:
```
router.head(
  [`/dist/:branch`, `/download/release/:branch`, /* ... */],
  [substitutionMiddleware]
)
```

Like a path parameter :)